### ISSUE TITLE
Infernal Doesn't Hurt Same-Faction Players When Summoned

### ISSUE THREAD

**Issue by YourNewfriend** [ccshiro/cc-buglist#856]
After paying a bit more attention to detail, I determined that the bug is that the Infernal on Corecraft's test realm isn't hostile for a moment before becoming enslaved automatically, and during this period of hostility, it should have enough time to swing its arm once at nearby players.

It may be that the Infernal should only be capable of harming PvP targets, as well as players in the same group as the summoner (like when it is released from slavery), but who knows.  Just do whatever's easier or something.

Patch 1.3: "Inferno - The area of effect damage caused by the Inferno spell, and all threat generated by it, will now originate from the Infernal. The Infernal will now immediately attack targets within the radius of the spell."

Evidence:
https://youtu.be/yva7MbQy11g?t=7m35s

Additional contributions are welcome.

---

[old]The process of an infernal dropping from the sky should hurt/stun group members as well as PvP flagged players.
I'm uncertain as to why the player on the right was unaffected.[/old]

**Comment by YourNewfriend** [comment]
In this video, the caster herself is injured.  She wasn't PvP flagged, but she could've had summoning threat.
https://youtu.be/BHdcekqLl_A?t=24s

**Comment by YourNewfriend** [comment]
![wowscrnshot_070615_030214](https://cloud.githubusercontent.com/assets/13057166/8520013/9268c0e0-238b-11e5-8589-38a8c3e96552.jpg)

That was an Inferno crit on CC's test server on a target with 0 resilience.  In the video, a level 54 Warlock's Inferno hits her for 425 damage, and a nearby critter for 200.  The damage appeared white in the video, but I was uncertain, but I believe I'm positive now.  

If you look carefully at the second video, you can see that it was in fact the Inferno assaulting its master with a white hit during its brief moment of hostility, and that's the only special same-faction-damaging feature that summoning an Infernal has, confirming the lingering suspicion I had after watching the original video I provided.

**Comment by ccshiro** [comment]
The videos are inconsistent, and the behavior is erratic. Basically we went with what we think is the intended behavior. We implemented it to work like:
- Inferno when he lands casts the AoE (not the warlock being the source as it was previously); this results in the warlock and any PvP flagged targets can be hit if they're within the area.
- Roughly a second later he gets enslaved by the warlock (basically, starts out hostile, could possibly get a swing off, then goes friendly).
